### PR TITLE
Fixed incorrect types when constructing VSMs in tests.

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

File: src/test/scala/is/hail/variant/vsm/VSMSuite.scala [hunk 1]
```diff
@@ -58,23 +58,19 @@ class VSMSuite extends SparkSuite {
...
       Annotation.emptyIndexedSeq(2),
       Annotation.empty,
-      TStruct(
+      sSignature = TStruct(
         "inner" -> TStruct(
           "thing1" -> TString),
-        "thing2" -> TString),
-      TStruct.empty,
-      TStruct.empty)
+        "thing2" -> TString))
```
Comment (jbloom22) [inline src/test/scala/is/hail/variant/vsm/VSMSuite.scala]: The `vaSignature` defaults to `TStruct.empty` but here the variant annotations have the form `Annotation(Annotation("yes"), "yes")`, so for consistency I think the type should be something like `TStruct("q1" -> TStruct("q2" -> TString))`.

Also, you may as well remove the variables `r1`, `r2` and `r3` below and just write `val va1 = Annotation(Annotation("yes"), "yes")`, etc.

**Comment by cseed** [comment]
Addressed comments.  That test was messed up.  Added typecheck and fixed everything up so it passes.

File: src/test/scala/is/hail/variant/vsm/VSMSuite.scala [hunk 2]
```diff
@@ -52,115 +52,140 @@ class VSMSuite extends SparkSuite {
     val vds2 = hc.importVCF("src/test/resources/sample.vcf.gz", force = true)
     assert(vds1.same(vds2))
 
-    val mdata1 = VSMFileMetadata(Array("S1", "S2", "S3"))
-    val mdata2 = VSMFileMetadata(Array("S1", "S2"))
-    val mdata3 = VSMFileMetadata(
+    val s1mdata = VSMFileMetadata(Array("S1", "S2", "S3"))
+    val s1va1: Annotation = null
+    val s1va2 = Annotation()
+    val s1va3 = Annotation()
```
Comment (jbloom22) [inline src/test/scala/is/hail/variant/vsm/VSMSuite.scala]: Never used.

File: src/test/scala/is/hail/variant/vsm/VSMSuite.scala [hunk 3]
```diff
@@ -52,115 +52,140 @@ class VSMSuite extends SparkSuite {
...
+
+    val s2mdata = VSMFileMetadata(Array("S1", "S2"))
+    val s2va1: Annotation = null
+    val s2va2: Annotation = null
+    val s2va3 = Annotation()
```
Comment (jbloom22) [inline src/test/scala/is/hail/variant/vsm/VSMSuite.scala]: Never used.

File: src/test/scala/is/hail/variant/vsm/VSMSuite.scala [hunk 4]
```diff
@@ -52,115 +52,140 @@ class VSMSuite extends SparkSuite {
...
+    val s1rdd7 = sc.parallelize(Seq((v1, (s1va1,
+      Iterable(Genotype(),
+        Genotype(0),
+        Genotype(2)))),
+      (v2, (s1va2,
+        Iterable(Genotype(0),
+          Genotype(0),
+          Genotype(1)))))).toOrderedRDD
+
+    val s2rdd8 = sc.parallelize(Seq((v1, (s2va1,
```
Comment (jbloom22) [inline src/test/scala/is/hail/variant/vsm/VSMSuite.scala]: Never used.  Did you intend to add `new VariantDataset(hc, s2mdata, s2rdd8)` to the list below?

File: src/test/scala/is/hail/variant/vsm/VSMSuite.scala [hunk 5]
```diff
@@ -52,115 +52,140 @@ class VSMSuite extends SparkSuite {
...
       Iterable(Genotype(),
         Genotype(1),
         Genotype(2)))),
-      (v2, (va2,
+      (v2, (s3va2,
         Iterable(Genotype(0),
           Genotype(0),
           Genotype(1)))))).toOrderedRDD
 
     // for mdata2
```
Comment (jbloom22) [inline src/test/scala/is/hail/variant/vsm/VSMSuite.scala]: `// for s2mdata`

File: src/test/scala/is/hail/variant/vsm/VSMSuite.scala [hunk 6]
```diff
@@ -52,115 +52,140 @@ class VSMSuite extends SparkSuite {
...
         Genotype(0),
         Genotype(0)))))).toOrderedRDD
 
     // differ in number of variants
-    val rdd5 = sc.parallelize(Seq((v1, (va1,
+    val s3rdd5 = sc.parallelize(Seq((v1, (s3va1,
       Iterable(Genotype(),
         Genotype(0)))))).toOrderedRDD
 
     // differ in annotations
```
Comment (jbloom22) [inline src/test/scala/is/hail/variant/vsm/VSMSuite.scala]: `// differ in variant annotations`

**Comment by cseed** [comment]
Addressed comments.

**Pull request merged**